frontend/src/app/(pages)/profile/page.tsx

```diff
@@ -1,9 +1,69 @@
+"use client"
 import ProfileComponent from "@/components/profile/Profile"
+import { UserService } from "@/helpers/user/user_api_wrappers";
+import User from "@/types/user";
+import { Metadata } from "next"
+import { FormEvent, useEffect, useState } from "react";
+import { toast } from "react-toastify";
 
-export default function ProfilePage() {
+export const metadata: Metadata = {
+    title: "Profile",
+    description: "profile information",
+}
+
+export default async function ProfilePage() {
+
+    const [user, setUser] = useState<User | undefined>(undefined);
+
+    const email = typeof window !== "undefined" ? sessionStorage.getItem("email") || "" : "";
+
+    useEffect(() => {
+        fetchUser();
+    }, [])
+
+    const fetchUser = async () => {
+        const rawUser = await UserService.getUserByEmail(email);
+        console.log("User retrieved: " + JSON.stringify(rawUser));
+        setUser(rawUser);
+    }
+    
+    async function saveInformation(e: FormEvent<HTMLFormElement>, updatedUser: User, preferences: Preference) {
+        e.preventDefault();
+        try {
+            // Redundant error handling for typescript to stop error
```

Comment:
If I understand correctly, this is not redundant. Because you defined your useState `user` as either `User` or undefined, so there is a chance that this `user` is undefined. We are checking it here to remove such possibility. There are 2 ways I can see to resolve this. One is by using your current implementation (and you can probably remove this comment xD), and another one is you declared `user` as `User`, then after you called `getUserById`, you check if the returned data is not undefined before you assign.